src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1568,6 +1576,21 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
                 ->setArguments(array($transport['dsn'], $transport['options']))
                 ->addTag('messenger.receiver', array('alias' => $name))
             ;
+
+            if (0 === strpos($transport['dsn'], 'symfony://kernel.terminate')) {
```

Comment:
Hmm. I just realized this won't work with env vars.
As I don't see any use-cases for it, I think it's fine, but we need extra pov.